apps/docs/src/app/core/component-docs/alert/alert-docs.component.ts

```diff
@@ -128,11 +137,25 @@ export class AlertDocsComponent {
 
     schema: Schema;
 
-    constructor(private schemaFactory: SchemaFactoryService) {
+    @ViewChild('alert')
+    alertComponent: AlertComponent;
+
+    constructor(private schemaFactory: SchemaFactoryService, private _changeDetectorRef: ChangeDetectorRef) {
         this.schema = this.schemaFactory.getComponent('alert');
     }
 
     onSchemaValues(data): void {
         this.data = data;
+        this.openDynamicAlert();
+    }
+
+    /** opens alert */
+    openDynamicAlert(): void {
+        this._changeDetectorRef.detectChanges();
```

Comment:
probably to be called after opening the alert, but not sure is it needed.

Reply:
oh, sorry. To make it more understandable, it should be after `this.data` change.

But now I found  an inconsistency in auto dismiss time on fast change in duration input. So I am bringing back the shouldShow as used before to enable alert-open on button click only (not on changing the data).